dht/dht.go

```diff
@@ -184,49 +206,42 @@ func (dht *distributedHashTable) InsertContent(hash id.Hash, contentType uint8,
 }
 
 // DeleteContent from the DHT.
-func (dht *distributedHashTable) DeleteContent(hash id.Hash) {
-	dht.contentResolver.Delete(hash)
+func (dht *distributedHashTable) DeleteContent(hash id.Hash, contentType uint8) {
+	dht.contentResolver.Delete(hash, contentType)
 }
 
 // Content returns the content associated with a hash. If there is no
 // associated content, it returns false. Otherwise, it returns true.
-func (dht *distributedHashTable) Content(hash id.Hash) ([]byte, bool) {
-	return dht.contentResolver.Content(hash)
+func (dht *distributedHashTable) Content(hash id.Hash, contentType uint8) ([]byte, bool) {
+	return dht.contentResolver.Content(hash, contentType, true)
 }
 
 // HasContent returns true when there is content associated with the given hash.
 // Otherwise, it returns false. This is more efficient for checking existence
 // than the Content method, because no bytes are copied.
-func (dht *distributedHashTable) HasContent(hash id.Hash) bool {
-	_, ok := dht.contentResolver.Content(hash)
+func (dht *distributedHashTable) HasContent(hash id.Hash, contentType uint8) bool {
+	_, ok := dht.contentResolver.Content(hash, contentType, false)
 	return ok
 }
 
 // HasContent returns true when there empty/nil content associated with the
 // given hash. Otherwise, it returns false. This is more efficiently for
 // checking existence than the Content method, because no bytes are copied.
 // Note: not having content is different from having empty/nil content.
-func (dht *distributedHashTable) HasEmptyContent(hash id.Hash) bool {
-	content, ok := dht.contentResolver.Content(hash)
+func (dht *distributedHashTable) HasEmptyContent(hash id.Hash, contentType uint8) bool {
+	content, ok := dht.contentResolver.Content(hash, contentType, false)
 	return ok && len(content) == 0
 }
 
 func (dht *distributedHashTable) AddSubnet(signatories []id.Signatory) id.Hash {
 	copied := make([]id.Signatory, len(signatories))
 	copy(copied, signatories)
+
+	// Sort signatories in order of their XOR distance from our own address.
 	sort.Slice(copied, func(i, j int) bool {
-		for b := 0; b < 32; b++ {
-			d1 := dht.identity[b] ^ copied[i][b]
-			d2 := dht.identity[b] ^ copied[j][b]
-			if d1 < d2 {
-				return true
-			}
-			if d2 < d1 {
-				return false
-			}
-		}
-		return false
+		return dht.isCloser(copied[i], copied[j])
 	})
+
 	hash := id.NewMerkleHashFromSignatories(copied)
```

Comment:
Merkle root should be computed based on the *unsorted* slice that was handed to the DHT, otherwise, everyone will have a different merkle root.

Reply:
This was the behaviour on the existing `release/0.4.0` branch but I can update it if it was unintentional.